crates/accelerate/src/synthesis/clifford/greedy_synthesis.rs

```diff
@@ -0,0 +1,707 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use indexmap::IndexSet;
+use ndarray::{azip, s, Array1, Array2, ArrayView2};
+use smallvec::{smallvec, SmallVec};
+
+use crate::synthesis::linear::utils::calc_inverse_matrix_inner;
+use qiskit_circuit::operations::{Param, StandardGate};
+use qiskit_circuit::Qubit;
+
+/// Symplectic matrix.
+/// Currently this class is internal to the greedy synthesis algorithm.
+struct SymplecticMatrix {
```

Comment:
I would suggest to move all this code (until line 232) to a helper file called clifford_utils.rs, which can be moved later to quantum_info when porting the Clifford class to rust (https://github.com/Qiskit/qiskit/issues/12367).
We can change the API later, but we should also keep in mind that these classes would later be used for the general Clifford class operations.